### PR TITLE
[Feature][Transforms-V2] llm trans support field projection

### PR DESCRIPTION
**Pull request by hawk9821**
### Purpose of this pull request
support for specifying which columns of incoming data should be used as input to the LLM, reducing token consumption and eliminating the impact of irrelevant data

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
UT:  LLMRequestJsonTest#testOpenAIProjectionRequestJson
E2E: TestLLMIT#testLLMWithOpenAIColumns

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/transform-v2/llm.md [hunk 1]
```diff
@@ -11,17 +11,18 @@ more.
 ## Options
 
 | name                   | type   | required | default value |
-|------------------------|--------|----------|---------------|
+| ---------------------- | ------ | -------- | ------------- |
 | model_provider         | enum   | yes      |               |
 | output_data_type       | enum   | no       | String        |
 | prompt                 | string | yes      |               |
+| inference_projection   | list   | no       |               |
```
Comment (Hisoka-X) [inline docs/en/transform-v2/llm.md]:
```suggestion
| inference_columns   | list   | no       |               |
```

Comment (hawk9821) [reply docs/en/transform-v2/llm.md]: get

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/llm/LLMTransform.java [hunk 2]
```diff
@@ -112,6 +115,8 @@ public void open() {
     protected Object getOutputFieldValue(SeaTunnelRowAccessor inputRow) {
         tryOpen();
         SeaTunnelRow seaTunnelRow = new SeaTunnelRow(inputRow.getFields());
+        TableSchema tableSchema = transformTableSchema();
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/llm/LLMTransform.java]: what's for?

Comment (hawk9821) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/llm/LLMTransform.java]: Forgot to delete,  fix

**Review by Hisoka-X — APPROVED** [review]

**Comment by Hisoka-X** [comment]
Thanks @hawk9821

**Review by hailin0 — APPROVED** [review]

**Pull request merged**